### DISCUSSION TITLE
Namespace Change

### DISCUSSION THREAD

**Discussion by kayhantolga** [betalgo/openai#188] [Polls]
I am planning to change the namespace from Betalgo.OpenAI.GPT3 to Betalgo.OpenAI. The reason for this change is quite obvious. OpenAI has recently released several new products, including Chat GPT, GPT-4, and Whisper Dall-E, and has decided to add these products to their current API offering. Therefore, it makes sense for us to update our namespace to Betalgo.OpenAI to reflect that we cover all of OpenAI's products, not just GPT-3. I would appreciate your feedback on this change. Please vote and share your opinion.

**Comment by Frogley** [comment]
Actually keeping it as `Betalgo.OpenAI.GPT3` will ruin my life because of my OCD.

**Comment by nilaoda** [comment]
It would be better if async functions could end with 'Async', otherwise users may mistakenly call them as synchronous functions.

`CreateCompletion` -> `CreateCompletionAsync`

**Reply by kayhantolga** [reply]
I am doing some research on this. Some libraries follow naming patterns without the "Async" suffix.

**Reply by nilaoda** [reply]
> I am doing some research on this. Some libraries follow naming patterns without the "Async" suffix.

Of course, it depends on your willingness. Thank you for your project!

**Comment by kayhantolga** [comment]
I don't think this namespace change will be easy, I am not sure if nuget allows me to change the package id. Instead, It release a new version.
